lib/instrumentation/core/http2.js

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2025 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const NAMES = require('../../metrics/names')
+const recordExternal = require('../../metrics/recorders/http_external')
+const logger = require('../../logger').child({ component: 'http2' })
+const synthetics = require('../../synthetics')
+const urltils = require('../../util/urltils.js')
+const cat = require('#agentlib/util/cat.js')
+module.exports = initialize
+
+function initialize(agent, http2, moduleName, shim) {
+  shim.wrap(http2, 'connect', wrapConnect)
+  shim.wrapReturn(http2, 'connect', wrapSession)
+  const moduleCtx = this
+
+  /**
+   * Wraps the http2 connect method in instrumentation, and saves the connect URL.
+   * The .request() method's argument doesn't retain the protocol of the connection,
+   * but it's available here as arg[0]
+   *
+   * @param {object} shim the generic shim with which the agent instruments
+   * @param {Function} fn the original connect function
+   * @returns {Function} an instrumented connect function
+   */
+  function wrapConnect(shim, fn) {
+    return function wrappedConnect(...args) {
+      moduleCtx.http2ConnectUrl = args[0]
```

Comment:
this should be a symbol

Reply:
also this assumes there's only one connection, which is not true. A better approach would be this:

```js
const result = fn.apply(this, args)
result[http2ConnectUrl] = args[0]
return result
```